exla/lib/exla/defn.ex

```diff
@@ -368,12 +368,13 @@ defmodule EXLA.Defn do
 
     {debug?, options} = Keyword.pop(options, :debug, false)
 
-    {args_key, reverse_args_triplet} =
+    {args_key, reverse_args_identifiers} =
       Enum.map_reduce(vars, [], fn var, acc ->
         Nx.Defn.Composite.traverse(var, acc, fn
-          %T{type: type, shape: shape, names: names}, acc ->
-            triplet = {type, shape, names}
-            {triplet, [triplet | acc]}
+          %T{} = t, acc ->
+            %T{type: type, shape: shape, names: names} = Nx.devectorize(t)
+            identifier = {type, shape, names}
+            {identifier, [identifier | acc]}
```

Comment:
I think we want the vectorized_axes to be part of the cache key (first argument) and the expanded one for the identifier. But you can also have return `{[identifier | vectorized_axes], [identifier | acc]}` and that would also work!